src/App.jsx

```diff
@@ -14,45 +15,114 @@ export const goodsFromServer = [
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_LENGTH = 'length';
+const SORT_ALPHABET = 'alphabet';
+const REVERSE_FLAG = 'reverse';
+
+export default function App() {
+  const [state, setState] = useState({
+    reverse: '',
+    sortField: '',
+    visibleGoods: [...goodsFromServer],
+  });
+
+  function reverseGoods() {
+    setState(prevState => ({
+      ...prevState,
+      reverse: prevState.reverse === REVERSE_FLAG ? '' : REVERSE_FLAG,
+      visibleGoods: [...prevState.visibleGoods].reverse(),
+    }));
+  }
+
+  function sortByLength() {
+    const sortedVisibleGoods = [...goodsFromServer]
+      .sort((a, b) => a.length - b.length);
+
+    if (state.reverse === REVERSE_FLAG) {
+      sortedVisibleGoods.reverse();
+    }
+
+    setState(prevState => ({
+      ...prevState,
+      sortField: SORT_LENGTH,
+      visibleGoods: sortedVisibleGoods,
+    }));
+  }
+
+  function sortByAlphabet() {
+    const sortedVisibleGoods = [...goodsFromServer]
+      .sort((a, b) => a.localeCompare(b));
+
+    if (state.reverse === REVERSE_FLAG) {
+      sortedVisibleGoods.reverse();
+    }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+    setState(prevState => ({
+      ...prevState,
+      sortField: SORT_ALPHABET,
+      visibleGoods: sortedVisibleGoods,
+    }));
+  }
+
+  function resetGoods() {
+    setState({
+      reverse: '',
+      sortField: '',
+      visibleGoods: [...goodsFromServer],
+    });
+  }
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={sortByAlphabet}
+          type="button"
+          className={`button is-info ${
+            state.sortField !== SORT_ALPHABET ? 'is-light' : ''
+          }`}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={sortByLength}
+          type="button"
+          className={`button is-success ${
+            state.sortField !== SORT_LENGTH ? 'is-light' : ''
+          }`}
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={reverseGoods}
+          type="button"
+          className={`button is-warning ${
+            state.reverse !== REVERSE_FLAG ? 'is-light' : ''
+          }`}
+        >
+          Reverse
+        </button>
+
+        {state.sortField !== '' || state.reverse !== '' ? (
```

Comment:
```suggestion
        {state.sortField || state.reverse ? (
```

and you can use && instead use ? and null.